index.js

```diff
@@ -119,12 +119,12 @@ class Plugin extends AbstractBtpPlugin {
           if (this._store) {
             await this._store.load(account + ':token')
             const storedToken = this._store.get(account + ':token')
-            if (storedToken && storedToken !== token) {
+            if (storedToken && storedToken !== tokenToAccount(token)) {
               throw new Error('incorrect token for account.' +
                 ' account=' + account +
                 ' token=' + token)
```

Comment:
Wouldn't it be impossible to get here if the account is stored instead of the token? If you only store the account, you'll either get a new account or one that already exists, but there's no way to have an invalid credential for an account.

Reply:
I added tests. Do you still have doubts that the code works as intended?